modules/gdscript/tests/scripts/analyzer/features/for_range_usage.gd

```diff
@@ -0,0 +1,7 @@
+func test():
+	var array := [3, 6, 9]
+	var result := ''
+	for i in range(array.size(), 0, -1):
+		result += str(array[i - 1])
+	assert(result == '963')
+	print('ok')
```

Comment:
would be good to increase the test coverage by more variations.

e.g. using also a negative value for argument two
```
func test_a():
	var data := PackedStringArray(["1","2","3","4"])
	var result := ""
	for index in range(data.size()-1, -1, -1):
		result += data[index]
	assert(result == '4321')
    print('ok')
```

Reply:
The crash happens because not all arguments are constants. There wasn't a single test with a non-constant argument. This test is about that.

There is [a test](https://github.com/godotengine/godot/blob/e930c8d3838280e40baabc4426bd8236f7ac50a3/modules/gdscript/tests/scripts/parser/features/for_range.gd#L18) with negative value for second argument.